expr/views_test.go

```diff
@@ -0,0 +1,644 @@
+	}{
+		{ehs1: &mSoftwareSystem, want: 1},
+		{ehs1: &mSoftwareSystem, ehs2: &mContainer, want: 1},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			if i == 0 {
+				mLandscapeView.AddElements(tt.ehs1)
+			} else {
+				mLandscapeView.AddElements(tt.ehs1, tt.ehs2)
+			}
+
+			got := len(mLandscapeView.ElementViews)
+			if got != tt.want {
+				t.Errorf("Got %d, wanted %d", got, tt.want)
+			}
+		})
+	}
+
+}
+
+func Test_AddElements2ContextView(t *testing.T) {
+	mLViewProp := ViewProps{Key: "LV"}
+	mContextView := ContextView{ViewProps: &mLViewProp}
+	mSoftwareSystem := SoftwareSystem{
+		Element: &Element{Name: "SoftwareSystem"},
+	}
+	mContainer := Container{
+		Element: &Element{Name: "Container"},
+	}
+	tests := []struct {
+		ehs1 ElementHolder
+		ehs2 ElementHolder
+		want int
+	}{
+		{ehs1: &mSoftwareSystem, want: 1},
+		{ehs1: &mSoftwareSystem, ehs2: &mContainer, want: 1},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			if i == 0 {
+				mContextView.AddElements(tt.ehs1)
+			} else {
+				mContextView.AddElements(tt.ehs1, tt.ehs2)
+			}
+
+			got := len(mContextView.ElementViews)
+			if got != tt.want {
+				t.Errorf("Got %d, wanted %d", got, tt.want)
+			}
+		})
+	}
+
+}
+
+func Test_AddElements2ContainerView(t *testing.T) {
+	mLViewProp := ViewProps{Key: "LV"}
+	mContainerView := ContainerView{ViewProps: &mLViewProp}
+	mSoftwareSystem := SoftwareSystem{
+		Element: &Element{Name: "SoftwareSystem"},
+	}
+	mComponent := Component{
+		Element: &Element{Name: "Component"},
+	}
+	tests := []struct {
+		ehs1 ElementHolder
+		ehs2 ElementHolder
+		want int
+	}{
+		{ehs1: &mSoftwareSystem, want: 1},
+		{ehs1: &mSoftwareSystem, ehs2: &mComponent, want: 1},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			if i == 0 {
+				mContainerView.AddElements(tt.ehs1)
+			} else {
+				mContainerView.AddElements(tt.ehs1, tt.ehs2)
+			}
+
+			got := len(mContainerView.ElementViews)
+			if got != tt.want {
+				t.Errorf("Got %d, wanted %d", got, tt.want)
+			}
+		})
+	}
+
+}
+
+func Test_AddElements2ComponentView(t *testing.T) {
+	mLViewProp := ViewProps{Key: "LV"}
+	mComponentView := ComponentView{ViewProps: &mLViewProp}
+	mSoftwareSystem := SoftwareSystem{
+		Element: &Element{Name: "SoftwareSystem"},
+	}
+	mComponent := Component{
+		Element: &Element{Name: "Component"},
+	}
+	tests := []struct {
+		ehs1 ElementHolder
+		ehs2 ElementHolder
+		want int
+	}{
+		{ehs1: &mSoftwareSystem, want: 1},
+		{ehs1: &mSoftwareSystem, ehs2: mComponent.Element, want: 1},
+	}
+	for i, tt := range tests {
+		t.Run(fmt.Sprint(i), func(t *testing.T) {
+			if i == 0 {
+				mComponentView.AddElements(tt.ehs1)
+			} else {
+				mComponentView.AddElements(tt.ehs1, tt.ehs2)
+			}
+
+			got := len(mComponentView.ElementViews)
+			if got != tt.want {
+				t.Errorf("Got %d, wanted %d", got, tt.want)
+			}
+		})
+	}
+
+}
+
+/*
+func Test_AddElements2DeploymentView(t *testing.T) {
```

Comment:
Might be good to remove this commented out code. Alternatively we could make it work or add a header comment explaining why it's commented out.

Reply:
I've added an issue as you suggested to explain why I've not been able to complete it so far. It's probably a lack of understanding on my part.